### DISCUSSION TITLE
GitHub Pages failing to build

### DISCUSSION THREAD

**Discussion by HaikuJock** [community/community#22807] [Other Feature Feedback, Questions, & Ideas]
<p>Any changes to GitHub Pages in the last week?</p>
<p>I add a new blog post to my site once per week. The site builds locally and looks fine. Today the GitHub Pages build started failing:</p>
<p>"The page build failed for the <code>master</code> branch with the following error:</p>
<p>Page build failed."</p>
<p>No more details on the commit actions.</p>
<p>Making a pull-request for changes does not provide any more information on the Checks tab of the PR (as suggested by <a href="https://docs.github.com/en/free-pro-team@latest/github/working-with-github-pages/about-jekyll-build-errors-for-github-pages-sites" rel="noopener nofollow ugc">https://docs.github.com/en/free-pro-team@latest/github/working-with-github-pages/about-jekyll-build-errors-for-github-pages-sites</a>)</p>
<p>Reverting the repository to last week’s commit and the site still does not build.<br>
</p><aside class="onebox allowlistedgeneric">
  <header class="source">
      <img src="https://github.githubassets.com/favicons/favicon.svg" class="site-icon" width="32" height="32">
      <a href="https://github.com/HaikuJock/haikujock.github.io" target="_blank" rel="noopener nofollow ugc">GitHub</a>
  </header>
  <article class="onebox-body">
    <img src="https://user-images.githubusercontent.com/594652/181073189-bbd3a5a4-e0fe-4752-86a9-ee16dcb6d9ac.png" class="thumbnail onebox-avatar" width="256" height="256">

<h3><a href="https://github.com/HaikuJock/haikujock.github.io" target="_blank" rel="noopener nofollow ugc">HaikuJock/haikujock.github.io</a></h3>

<p>Dwerg Saga Website. Contribute to HaikuJock/haikujock.github.io development by creating an account on GitHub.</p>

  </article>
  <div class="onebox-metadata">
    
    
  </div>
  <div style="clear: both"></div>
</aside>

<p>Thank you for your time.</p>

**Comment by HaikuJock** [comment]
<p>Adding <code>gem 'github-pages'</code> to my projects Gemfile under <code>group :jekyll_plugins do</code> gives me a new error when I build locally:</p>
<pre><code class="lang-auto">Liquid Exception: undefined method `encode' for #&lt;Jekyll::GitHubMetadata::Value:0x000000000822cc90&gt; in /_layouts/default.html
</code></pre>
<p>No line number so I’ll be binary-deleting to find it…</p>

**Comment by HaikuJock** [comment]
<p>I think this is a dark alley and a grue lives here</p>

**Comment by HaikuJock** [comment]
<pre><code class="lang-auto">    {% feed_meta %}
</code></pre>
<p>is the offender…</p>

**Comment by HaikuJock** [comment, marked as answer]
<pre><code class="lang-auto">    {% feed_meta %}
</code></pre>
<p>apparently generates <code>sitemap.xml</code><br>
</p><aside class="onebox allowlistedgeneric">
  <header class="source">
      <img src="https://user-images.githubusercontent.com/594652/181073192-58e887e0-572a-4806-b86b-1f50ef2f711e.png" class="site-icon" width="32" height="32">
      <a href="https://brianm.me/posts/how-to-make-jekyll-site-blog#feed" target="_blank" rel="noopener nofollow ugc" title="12:00AM - 01 September 2016">Brian Mitchell – 1 Sep 16</a>
  </header>
  <article class="onebox-body">
    <img src="" class="thumbnail" width="" height="">

<h3><a href="https://brianm.me/posts/how-to-make-jekyll-site-blog#feed" target="_blank" rel="noopener nofollow ugc">How to Make a Jekyll Site/Blog</a></h3>

<p>A tutorial on making a Jekyll based site/blog.</p>


  </article>
  <div class="onebox-metadata">
    
    
  </div>
  <div style="clear: both"></div>
</aside>

<p>My Gemfile is now just</p>
<pre><code class="lang-auto">gem "github-pages", "~&gt; 208", group: :jekyll_plugins
</code></pre>
<p>and I’ve removed the <code>{% feed_meta %}</code> from my <code>_layouts/default.html</code> and the <code>sitemap.xml</code> is still generated. Looks legit.<br>
I’m also running <code>bundle exec jekyll serve</code> from MinGW32 rather than Windows.</p>

**Comment by shurrey** [comment]
<p>Thanks <a class="mention" href="/haikujock">@HaikuJock</a>. This was kicking my rear. LOL I changed a placeholder link to a real link and couldn’t get it to build. Spent a day mucking with dependencies and was about to delete my repo. LOL</p>
<p>Have a :beer: on me, my friend.</p>
